controllers/managedosversionchannel_controller.go

```diff
@@ -375,6 +375,13 @@ func (r *ManagedOSVersionChannelReconciler) createManagedOSVersions(ctx context.
 				logger.Error(err, "Could not patch ManagedOSVersion as no longer in sync", "name", version.Name)
 				return fmt.Errorf("deprecating ManagedOSVersion '%s': %w", version.Name, err)
 			}
+			if ch.Spec.DeleteNoLongerInSyncVersions {
```

Comment:
Having the annotation however is still nice to have, so it will show also in the UI as long the resource still exists.

Consider a finalizer is going to be added to the ManagedOSVersion (other PR), so this is just scheduling the resource for deletion, the finalizer could hold it indefinitely (as far it's still referenced by a ManagedOSImage).